doc/languages-frameworks/javascript.section.md

```diff
@@ -594,6 +594,24 @@ To install the package `yarnInstallHook` uses both `npm` and `yarn` to cleanup p
 
 - `yarnKeepDevDeps`: Disables the removal of devDependencies from `node_modules` before installation.
 
+#### Use with `yarn-berry` and newer lockfiles {#javascript-yarnberry}
+
+The above works well with `yarn` version 1. However, `yarn-berry` introduced a new version of lockfiles, which will not work with classic `yarn`. To still be able to use `fetchYarnDeps` with `yarn-berry`, you will need to supply the source folder containing the `yarn.lock`  and the corresponding project files, and the version of the lockfile:
+
+```nix
+yarnOfflineCache = fetchYarnDeps {
+    src = "${src}/web";
+    yarnVersion = 3;
+    hash = "...";
+  };
+```
+
+`yarnVersion = 3` is for `yarn-berry_3` lockfiles and `yarnVersion = 4` will use `yarn-berry_4` to generate `node_modules`.
+
+The hook to be used with `yarn-berry` is called `yarnBerry3ConfigHook` for version 3 lockfiles and `yarnBerry4ConfigHook` for version 4 lockfiles.
+
+There is currently neither an `yarnBuildHook` nor an `yarnInstallHook` for `yarn-berry`.
```

Comment:
And classic yarn v1 doesn't fit such projects? Perhaps that should be slightly more clear?

Reply:
I'm not sure what you mean. If upstream uses yarn-v1, we can use the usual hooks. It upstream uses a yarn-berry lockfile (and package.json) we'll have to use either the `yarnBerry3ConfigHook` or the `yarnBerry4ConfigHook`.

Reply:
got it. Done